tools/ports/sqlite3.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2022 The Emscripten Authors.  All rights reserved.
+# Emscripten is available under two separate licenses, the MIT license and the
+# University of Illinois/NCSA Open Source License.  Both these licenses can be
+# found in the LICENSE file.
+
+import os
+import shutil
+import logging
+
+TAG = '3380500'
+HASH = '4fc2992e4c1ca1664a9b01e07c9b944003c4ed0612978e471eff262a7114e4a0699244d91e71ae4ad2b5e1fc9829917cd7d5f0313aa5859036905f974548d94a'
+
+deps = []
+
+
+def needed(settings):
+  return settings.USE_SQLITE3
+
+
+def get_lib_name(settings):
+  return 'libsqlite3' + ('-mt' if settings.USE_PTHREADS else '') + '.a'
+
+
+def get(ports, settings, shared):
+  # TODO: This is an emscripten-hosted mirror of the sqlite repo from sqlite.prg.
+  ports.fetch_project('sqlite3', 'https://www.sqlite.org/2022/sqlite-amalgamation-' + TAG + '.zip', 'sqlite-amalgamation-' + TAG, sha512hash=HASH)
```

Comment:
Good point! So far I have not been able to locate a more suitable mirror for the release. The SQLite amalgamation release is a convenient single file bundle of SQlite (one C file, two header files). It would take more effort to build SQLite from the github source mirror of SQlite. Could Emscripten provide its own mirror, e.g. on emscripten-ports?

Reply:
Given how easy it is to integrate sqlite (especially the amalgamation version) into a project.. how much value is there in emscripten adding this port?  Can't folks who want to just use it?   I suppose the configuration options are non-trivial?